cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -59,13 +65,14 @@ func (t *GceTemplateBuilder) getAcceleratorCount(accelerators []*gce.Accelerator
 	return count
 }
 
-// BuildCapacity builds a list of resource capacities for a node.
-func (t *GceTemplateBuilder) BuildCapacity(machineType string, accelerators []*gce.AcceleratorConfig, zone string, cpu int64, mem int64) (apiv1.ResourceList, error) {
+// BuildCapacity builds a list of resource capacities given list of hardware.
+func (t *GceTemplateBuilder) BuildCapacity(cpu int64, mem int64, accelerators []*gce.AcceleratorConfig) (apiv1.ResourceList, error) {
 	capacity := apiv1.ResourceList{}
 	// TODO: get a real value.
 	capacity[apiv1.ResourcePods] = *resource.NewQuantity(110, resource.DecimalSI)
 	capacity[apiv1.ResourceCPU] = *resource.NewQuantity(cpu, resource.DecimalSI)
-	capacity[apiv1.ResourceMemory] = *resource.NewQuantity(mem, resource.DecimalSI)
+	memTotal := mem - CalculateCapacityFromPhysicalMemory(mem)
```

Comment:
IIUC, this is calculating _reservation_ and assigning it to memTotal. Is it intended?

Reply:
it was not, I have fixed that and added additional unit test